src/jit/compiler.h

```diff
@@ -7462,7 +7462,10 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
             return SIMD_AVX2_Supported;
         }
 
-        if (CanUseSSE4())
+        // SIMD_SSE4_Supported actually requires all of SSE3, SSSE3, SSE4.1, and SSE4.2
+        // to be supported. We can only enable it if all four are enabled in the compiler
+        if (compSupports(InstructionSet_SSE42) && compSupports(InstructionSet_SSE41) &&
```

Comment:
We could create a `compSupportsAll(InstructionSet_SSE3 | InstructionSet_SSSE3 | InstructionSet_SSE41 | InstructionSet_SSE42)` function to simplify this, if we think it is worthwhile.